sylt-compiler/src/typechecker.rs

```diff
@@ -1464,13 +1496,28 @@ impl TypeChecker {
     }
 
     fn solve(&mut self, statements: &Vec<(&Statement, usize)>) -> Result<(), Vec<Error>> {
+        // Initialize the namespaces first.
         for (statement, namespace) in statements.iter() {
-            self.outer_statement(
-                statement,
-                TypeCtx {
-                    namespace: *namespace,
-                },
-            )?;
+            if matches!(statement.kind, StatementKind::Use { .. }) {
+                self.outer_statement(
+                    statement,
+                    TypeCtx {
+                        namespace: *namespace,
+                    },
+                )?;
```

Comment:
Maybe move in the Use code here - and remove it from outer-statement?
Then the `matches!` would disapear. Which makes me warm and fuzzy inside!